app/jobs/socure_reason_code_download_job.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class SocureReasonCodeDownloadJob < ApplicationJob
+  include JobHelpers::StaleJobHelper
+
+  queue_as :low
+
+  discard_on JobHelpers::StaleJobHelper::StaleJobError
+
+  def perform
+    return unless IdentityConfig.store.idv_socure_reason_code_download_enabled
+
+    result = Proofing::Socure::ReasonCodes::Importer.new.synchronize
+    analytics.idv_socure_reason_code_download(**result.to_h)
+  end
+
+  def analytics
+    Analytics.new(
+      user: AnonymousUser.new,
+      request: nil,
+      sp: service_provider_issuer,
```

Comment:
I don't understand where `service_provider_issuer` is coming from. Most usages I'm seeing are reading it off a ServiceProvider record, or passing it through `perform`.

I am slightly concerned that this might be calling a non-existent method, and it's masked by using `FakeAnalytics` in the spec. But it is also extremely possible that I'm just missing the definition of this.

Reply:
Good catch! This did not get caught in the tests because we mock analytics everywhere. It is an artifact of copy/paste.

Reply:
I pushed d0ca99cabf566c7343d31a0ebf4774b896da854d to address this. I modified the test there so there is an example that doesn't have an expectation build around analytics. It fails without the corresponding change to the job. PTAL!